pkg/util/push/otel.go

```diff
@@ -175,7 +175,7 @@ func promToMimirTimeseries(promTs *prompb.TimeSeries) mimirpb.PreallocTimeseries
 
 	histograms := make([]mimirpb.Histogram, 0, len(promTs.Histograms))
 	for _, histogram := range promTs.Histograms {
-		histograms = append(histograms, promToMimirHistogram(histogram))
+		histograms = append(histograms, promToMimirHistogram(&histogram))
```

Comment:
This is half-way there. It still makes a copy into `histogram` local variable.

Reply:
Addressed that, but I am wondering whether that is going to be optimized by the compiler already.